### DISCUSSION TITLE
How can I get a reasonable value of acceptance-fractions

### DISCUSSION THREAD

**Discussion by YmcGW** [phoebe-project/phoebe2#684] [Questions]
Hi!
I tried to run MCMC of three datasets to get the uncertainty of some parameters, I found that the value of 'acceptance-fractions' was too small after checking the convergence, and the reasonable range given by the literature  **Foreman-Mackey et al. (2013)** was about 0.2-0.5. and This corner plot is not quite what I expected. The distribution of points is a little bit discrete. I'm sure my parameter range Settings are reasonable and the value of 'ntriangles' was 3000. Can you give me some advice to improve the value of 'acceptance-fractions' and corner plot. 

my script:

b.add_distribution({'q': phoebe.gaussian_around(0.01),
                    'fillout_factor': phoebe.gaussian_around(0.03),
                    'incl@binary': phoebe.gaussian_around(8),
                    'teff@secondary': phoebe.gaussian_around(600),
                    'pblum@primary@lcB': phoebe.gaussian_around(2),
                    'pblum@primary@lcV': phoebe.gaussian_around(2),
                    'pblum@primary@lcR': phoebe.gaussian_around(2)},                   
                    distribution='ball01')

b.add_solver('sampler.emcee',
             init_from='ball01',
             compute='phoebe01',
             solver='emcee_solver',overwrite=True)

b.run_solver('emcee_solver', niters=1000, nwalkers=20, solution='emcee_sol',overwrite=True)

I also tried 2000 and 3000 niters, but still no improvement.

![acceptance-fractions](https://user-images.githubusercontent.com/118886353/213874115-1b658384-c3d6-4680-a56c-23d7999e06b9.png)

![corner plot](https://user-images.githubusercontent.com/118886353/213874126-07e5bc1b-0253-4710-bf67-4ed3a42b043a.png)

![emcee diagnostic plot](https://user-images.githubusercontent.com/118886353/213874135-1f7407f7-61e1-4833-97dc-3c65c5e1a3a5.png)

**Comment by kecnry** [comment]
I'm guessing you'll likely need quite a few more iterations.  What does the log probability vs iteration plot look like (is the solution still improving or is the value of 101 really sufficient to remove the burnin time)?

**Reply by YmcGW** [reply]
I'm sorry that I can only provide this incomplete picture because the original picture was not saved. I tried the data of many binaries with similar mass ratios, and only the MCMC of this binary's data came out like this. All the other binaries were normal. 

![log probability vs iteration 1](https://user-images.githubusercontent.com/118886353/214083676-dae482e5-2e0d-4491-9422-6307e228f85b.jpg)
![log probability vs iteration 2](https://user-images.githubusercontent.com/118886353/214083775-d0952816-d9a1-4ef1-a5a8-8ce028dabc50.jpg)

**Comment by YmcGW** [comment]
If I want to increase the number of iterations of MCMC, do you suggest how many should be appropriate?

**Reply by kecnry** [reply]
My general advice would be to make sure that the solution has fully converged by updating the burnin parameter to exclude the iterations when the lnprobability is still trending upwards until the plot is generally flat but with walkers that are still exploring the parameter space.  You can always [continue a run from the end of a previous run](http://phoebe-project.org/docs/latest/tutorials/emcee_continue_from) to add more iterations, or if using the [servers infrastructure](http://phoebe-project.org/docs/latest/tutorials/server), you can keep the job running and monitor the results.

There was also a talk/tutorial near the end of this [past summer's workshop](http://phoebe-project.org/workshops/2022june/materials) called "Interpreting Chains and Convergence in MCMC" that might be applicable for you.

**Reply by YmcGW** [reply]
I found it hard to converge on the ' _f_ '. If this binary  is a semi-detached binary, but I use the contact binary model to run MCMC, will it also result in a bad corner plot? 
The light curve of this binary is hard to tell what type of binary it is., and its Roche radius is very close to the critical Roche radius.

**Reply by gecheline** [reply]
One way you can test if this is most likely to be a semi-detached binary is if the posterior of fillout factor is converging towards the 0-value edge. If this is the case then the contact binary model won’t work and it could be the culprit for the bad corner plot. I suggest you try running a detached model as well and see if you get a better fit with it. I’m suggesting a detached instead of a semi-detached because with the semi detached model you fix one of the radii to the critical radius and there’s no option to explore its posterior by changing the radius, so that will (artificially) affect the other parameters (mainly q and sma if you’re fitting for them).
Other options you can try for improving the corner plot is releasing some of the other fixed parameters, like the period, time of superior conjunction, temperatures of both stars, the noise nuisance parameters etc. I can’t tell from the plot whether you have all of these included already but as a general note it’s always worth exploring beyond the main light curve parameters to make sure that the assumed values of some of the other parameters are okay.
